android-core/src/main/kotlin/com/mparticle/internal/MPLifecycleCallbackDelegate.kt

```diff
@@ -0,0 +1,44 @@
+package com.mparticle.internal
+
+import android.annotation.TargetApi
+import android.app.Activity
+import android.app.Application.ActivityLifecycleCallbacks
+import android.os.Build
+import android.os.Bundle
+
+/**
+ * This class is used by the AppStateManager to determine when the app is visible or in the background.
+ *
+ * Separated into it's own class to avoid annoying logcat messages on pre-ICS devices.
+ */
+@TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
+internal class MPLifecycleCallbackDelegate(private val mStateManager: AppStateManager) :
+    ActivityLifecycleCallbacks {
+    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {
+        mStateManager.onActivityCreated(activity, savedInstanceState)
+    }
+
+    override fun onActivityStarted(activity: Activity) {
+        mStateManager.onActivityStarted(activity)
+    }
+
+    override fun onActivityResumed(activity: Activity) {
+        mStateManager.onActivityResumed(activity)
+    }
+
+    override fun onActivityPaused(activity: Activity) {
+        mStateManager.onActivityPaused(activity)
+    }
+
+    override fun onActivityStopped(activity: Activity) {
+        mStateManager.onActivityStopped(activity)
+    }
+
+    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {
```

Comment:
yes because this method are from ActivityLifecycleCallbacks interface and there:
  void onActivityCreated(@NonNull Activity activity, @Nullable Bundle savedInstanceState); 
   void onActivitySaveInstanceState(@NonNull Activity activity, @NonNull Bundle outState);
   
In this interface, `onActivityCreated` has a `savedInstanceState` that can be null, while `onActivitySaveInstanceState` requires a non-null `outState`.